aws-janitor/resources/route53.go

```diff
@@ -96,7 +96,9 @@ func route53ResourceRecordSetsForZone(opts Options, logger logrus.FieldLogger, s
 			}
 			logger.Warningf("%s: deleting %T: %s", o.ARN(), rrs, *rrs.Name)
 			if !opts.DryRun {
-				toDelete = append(toDelete, o)
+				if _, ok := opts.SkipResourceRecordSetTypes[*rrs.Type]; !ok{
```

Comment:
I have a feeling that we are practically trying to un-hardcode `resourceRecordSetIsManaged` by this change, is there some better way of telling which records are managed / "non-required" than just comparing their type?

Also, if reason why we adding this change is to better select managed records, should condition go inside `resourceRecordSetIsManaged`? Does `SkipRoute53ManagementCheck` make sense in context of what we've learned with NS and SOA records?

Reply:
> Does SkipRoute53ManagementCheck make sense in context of what we've learned with NS and SOA records?

Taking this back, looks like this is also suppressing regex matching which makes sense in either case. Though "management check" for records seem to mix "records that are managed because they look like something created by tests" and "required records that every zone should have". Should effect of SkipRoute53ManagementCheck apply only to former, but not latter? Should we have check for the latter active regardless of SkipRoute53ManagementCheck?

Reply:
1. function: zoneIsManaged only accepts dns zones with given name formats. For our usage, we need to skip that check since many of our dns zones that need to be handled by janitor don't have such kind of name formats.

2. SkipResourceRecordSetTypes filters out resource record sets with type: "NS" and "SOA", the reason why we need to so is that at line 182, when trying to delete resource record sets, the deletion fail with error log:
"unable to delete DNS records: InvalidChangeBatch: [A HostedZone must contain at least one NS record for the zone itself., A HostedZone must contain exactly one SOA record.]" 
which means that NS and SOA record sets should not be deleted

Reply:
Semantics we are getting just feels a bit confusing, "managed check" for records combines both type and name validation and type validation also happens to have important side effect of making deletions actually valid, and adding separate "skip resource record set types" adds one more place where type verification is happening.

I would imagine the following flags as potentially more flexible arrangement instead of complete managed check disabling and additional type check:

 - string list for managed zone regexes (defaulted to test-cncf-aws.k8s.io.)
 - string list for managed record regexes (defaulted to contents of managedNameRegexes)
 - string list for record types considered for deletion (defaulted to A)

Though if we already went ahead with disabling managed check altogether and no concerns were raised in previous reviews, I'm probably overthinking it and we can keep it as is.